pilot/pkg/security/authn/factory/factory.go

```diff
@@ -25,8 +25,12 @@ import (
 // for the given service instance.
 func NewPolicyApplier(push *model.PushContext,
 	serviceInstance *model.ServiceInstance, namespace string, labels labels.Collection) authn.PolicyApplier {
-	service := serviceInstance.Service
-	port := serviceInstance.ServicePort
+	var service *model.Service
+	var port *model.Port
+	if serviceInstance != nil {
+		service = serviceInstance.Service
+		port = serviceInstance.ServicePort
+	}
 	authnPolicy, _ := push.AuthenticationPolicyForWorkload(service, port)
```

Comment:
Can we check service, port here to set authnPolicy, instead of change the `AuthenticationPolicyForWorkload` implementation. That function should be deprecated once we remove alpha API (feel free to add comment for that for future reference).

You can also add another `NewPolicyApplier` that only take push-context,  namespace and labels, and may be a boolean for `excludeRequestAuthn` so that we don't need to do extra work for JWT when it's not needed. The current `NewPolicyApplier` will also be deprecated after removing alpha API.

Reply:
done